watertap/property_models/unit_specific/activated_sludge/asm3_reactions.py

```diff
@@ -0,0 +1,998 @@
+                "length": pyo.units.m,
+                "mass": pyo.units.kg,
+                "amount": pyo.units.kmol,
+                "temperature": pyo.units.K,
+            }
+        )
+
+
+class ASM3ReactionScaler(CustomScalerBase):
+    """
+    Scaler for the Activated Sludge Model No.1 reaction package.
+
+    Variables are scaled by their default scaling factor (if no user input provided), and constraints
+    are scaled using the inverse maximum scheme.
+    """
+
+    # TODO: Revisit this scaling factor
+    DEFAULT_SCALING_FACTORS = {"reaction_rate": 1e2}
+
+    def variable_scaling_routine(
+        self, model, overwrite: bool = False, submodel_scalers: dict = None
+    ):
+
+        if model.is_property_constructed("reaction_rate"):
+            for j in model.reaction_rate.values():
+                self.scale_variable_by_default(j, overwrite=overwrite)
+
+    def constraint_scaling_routine(
+        self, model, overwrite: bool = False, submodel_scalers: dict = None
+    ):
+        # TODO: Revisit this scaling methodology
+        # Consider scale_constraint_by_default or scale_constraints_by_jacobian_norm
+        if model.is_property_constructed("rate_expression"):
+            for j in model.rate_expression.values():
+                self.scale_constraint_by_nominal_value(
+                    j,
+                    scheme=ConstraintScalingScheme.inverseMaximum,
+                    overwrite=overwrite,
+                )
+
+
+class _ASM3ReactionBlock(ReactionBlockBase):
+    """
+    This Class contains methods which should be applied to Reaction Blocks as a
+    whole, rather than individual elements of indexed Reaction Blocks.
+    """
+
+    default_scaler = ASM3ReactionScaler
+
+    def initialize(self, outlvl=idaeslog.NOTSET, **kwargs):
+        """
+        Initialization routine for reaction package.
+
+        Keyword Arguments:
+            outlvl : sets output level of initialization routine
+
+        Returns:
+            None
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+        init_log.info("Initialization Complete.")
+
+
+@declare_process_block_class("ASM3ReactionBlock", block_class=_ASM3ReactionBlock)
+class ASM3ReactionBlockData(ReactionBlockDataBase):
+    """
+    ReactionBlock for ASM3.
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super().build()
+
+        # Create references to state vars
+        # Concentration
+        add_object_reference(self, "conc_mass_comp_ref", self.state_ref.conc_mass_comp)
+
+    # Rate of reaction method
+    def _rxn_rate(self):
+        self.reaction_rate = pyo.Var(
+            self.params.rate_reaction_idx,
+            initialize=0,
+            doc="Rate of reaction",
+            units=pyo.units.kg / pyo.units.m**3 / pyo.units.s,
+        )
+
+        theta_T_k_H = pyo.log(self.params.k_H["10C"] / self.params.k_H["20C"]) / (
+            10 - 20
```

Comment:
These 10 and 20 values are what I was referring to on our dev call. Technically, we can make these Params.

Reply:
@adam-a-a finisehd, add parameters ref_temp_1 and ref_temp_2